File: backend/posts/post.js
const express = require("express");
const router = express.Router();
const Post = require("../models/Posts");

//Get All posts
router.get("/", async (req, res) => {
  try {
    const post = await Post.find();
    res.json(post);
  } catch (err) {
    res.json(err);
  }
});

///Delete single post
router.delete("/:postId", async (req, res) => {
  try {
    const deletedPost = await Post.deleteOne({
      _id: req.params.postId,
    });
    res.json(deletedPost);
  }
  catch (err) {
    res.json(err);
  }
});

//Edit post we will edit title and body both
router.patch("/:postId", async (req, res) => {
  try {
    const updatedPost = await Post.updateOne(
      { _id: req.params.postId },
      { $set: { title: req.body.title, body: req.body.body } }
    );
    res.json(updatedPost);
  } catch (err) {
    res.json(err);
  }
});

//Submit post
router.post("/", async (req, res) => {
  const post = new Post({
    title: req.body.title,
    body: req.body.body,
    date: Date.now(),
  });
  try {
    const addedUser = await post.save();
    res.json(addedUser);
  } catch (err) {
    res.json(err);
  }
});

module.exports = router;

File: blogapp/lib/model/post.dart
class Post {
  String sId;
  String title;
  String body;
  String date;
  int iV;

  Post({this.sId, this.title, this.body, this.date, this.iV});

  Post.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    title = json['title'];
    body = json['body'];
    date = json['date'];
    iV = json['__v'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['title'] = this.title;
    data['body'] = this.body;
    data['date'] = this.date;
    data['__v'] = this.iV;
    return data;
  }
}
